ArchiverLib/Sources/ContentExtractorStore/ContentExtractorCache.swift

```diff
@@ -0,0 +1,183 @@
+            return nil
+        }
+    }
+
+    /// Save extraction result to cache
+    /// - Parameter entry: The cache entry to save
+    func saveCacheEntry(_ entry: CacheEntry) {
+        guard isEnabled else { return }
+
+        let cacheURL = cacheFileURL(for: entry.documentId)
+
+        do {
+            let data = try JSONEncoder().encode(entry)
+            try data.write(to: cacheURL, options: .atomic)
+            Logger.contentExtractor.debug("Cache entry saved for document ID: \(entry.documentId)")
+        } catch {
+            Logger.contentExtractor.error("Failed to save cache entry for document ID \(entry.documentId): \(error)")
+        }
+    }
+
+    /// Clear all cache entries
+    func clearCache() {
+        do {
+            let contents = try FileManager.default.contentsOfDirectory(at: cacheDirectory,
+                                                                       includingPropertiesForKeys: nil)
+            for fileURL in contents {
+                try FileManager.default.removeItem(at: fileURL)
+            }
+            Logger.contentExtractor.info("Cache cleared successfully")
+        } catch {
+            Logger.contentExtractor.error("Failed to clear cache: \(error)")
+        }
+    }
+
+    /// Get the number of cache entries
+    /// - Returns: The count of cached entries
+    func getCacheCount() -> Int {
+        do {
+            let contents = try FileManager.default.contentsOfDirectory(at: cacheDirectory,
+                                                                       includingPropertiesForKeys: nil)
+            return contents.count
+        } catch {
+            Logger.contentExtractor.error("Failed to get cache count: \(error)")
+            return 0
+        }
+    }
+
+    /// Get all cached document IDs
+    /// - Returns: Set of document IDs that have cache entries
+    func getCachedDocumentIds() -> Set<Document.ID> {
+        do {
+            let contents = try FileManager.default.contentsOfDirectory(at: cacheDirectory,
+                                                                       includingPropertiesForKeys: nil)
+            var ids = Set<Document.ID>()
+            for fileURL in contents {
+                if let entry = try? JSONDecoder().decode(CacheEntry.self, from: Data(contentsOf: fileURL)) {
+                    ids.insert(entry.documentId)
+                }
+            }
+            return ids
+        } catch {
+            Logger.contentExtractor.error("Failed to get cached document IDs: \(error)")
+            return []
+        }
+    }
+
+    /// Remove cache entries that don't have matching documents in the provided set
+    /// - Parameter validIds: Set of valid document IDs to keep
+    func pruneCache(keepingOnly validIds: Set<Document.ID>) {
+        do {
+            let contents = try FileManager.default.contentsOfDirectory(at: cacheDirectory,
```

Comment:
Just delete the complete folder or the contents in it. Do not decode JSON. Get documentId from filename